op-chain-ops/opbnb-upgrades/check.go

```diff
@@ -0,0 +1,899 @@
+	}
+	if newL2OutputOracleInfo.L2BlockTime != oldL2OutputOracleInfo.L2BlockTime {
+		return fmt.Errorf("L2OutputOracle L2BlockTime var check diff, expect %s actual %s", oldL2OutputOracleInfo.L2BlockTime, newL2OutputOracleInfo.L2BlockTime)
+	}
+	if newL2OutputOracleInfo.Challenger != oldL2OutputOracleInfo.Challenger {
+		return fmt.Errorf("L2OutputOracle Challenger var check diff, expect %s actual %s", oldL2OutputOracleInfo.Challenger, newL2OutputOracleInfo.Challenger)
+	}
+	if newL2OutputOracleInfo.Proposer != oldL2OutputOracleInfo.Proposer {
+		return fmt.Errorf("L2OutputOracle Proposer var check diff, expect %s actual %s", oldL2OutputOracleInfo.Proposer, newL2OutputOracleInfo.Proposer)
+	}
+	if newL2OutputOracleInfo.FinalizationPeriodSeconds != oldL2OutputOracleInfo.FinalizationPeriodSeconds {
+		return fmt.Errorf("L2OutputOracle FinalizationPeriodSeconds var check diff, expect %s actual %s", oldL2OutputOracleInfo.FinalizationPeriodSeconds, newL2OutputOracleInfo.FinalizationPeriodSeconds)
+	}
+	if newL2OutputOracleInfo.StartingBlockNumber != oldL2OutputOracleInfo.StartingBlockNumber {
+		return fmt.Errorf("L2OutputOracle StartingBlockNumber var check diff, expect %s actual %s", oldL2OutputOracleInfo.StartingBlockNumber, newL2OutputOracleInfo.StartingBlockNumber)
+	}
+	if newL2OutputOracleInfo.StartingTimestamp != oldL2OutputOracleInfo.StartingTimestamp {
+		return fmt.Errorf("L2OutputOracle StartingTimestamp var check diff, expect %s actual %s", oldL2OutputOracleInfo.StartingTimestamp, newL2OutputOracleInfo.StartingTimestamp)
+	}
+
+	// compare OptimismMintableERC20Factory
+	if newOptimismMintableERC20FactoryInfo.Bridge != oldOptimismMintableERC20FactoryInfo.Bridge {
+		return fmt.Errorf("OptimismMintableERC20Factory Bridge var check diff, expect %s actual %s", oldOptimismMintableERC20FactoryInfo.Bridge, newOptimismMintableERC20FactoryInfo.Bridge)
+	}
+
+	// compare OptimismPortal
+	if newOptimismPortalInfo.L2Oracle != oldOptimismPortalInfo.L2Oracle {
+		return fmt.Errorf("OptimismPortal L2Oracle var check diff, expect %s actual %s", oldOptimismPortalInfo.L2Oracle, newOptimismPortalInfo.L2Oracle)
+	}
+	if newOptimismPortalInfo.L2Sender != oldOptimismPortalInfo.L2Sender {
+		return fmt.Errorf("OptimismPortal L2Sender var check diff, expect %s actual %s", oldOptimismPortalInfo.L2Sender, newOptimismPortalInfo.L2Sender)
+	}
+	if newOptimismPortalInfo.SystemConfig != oldOptimismPortalInfo.SystemConfig {
+		return fmt.Errorf("OptimismPortal L2Oracle var check diff, expect %s actual %s", oldOptimismPortalInfo.L2Oracle, newOptimismPortalInfo.L2Oracle)
+	}
+	if newOptimismPortalInfo.Guardian != oldOptimismPortalInfo.Guardian {
+		return fmt.Errorf("OptimismPortal L2Oracle var check diff, expect %s actual %s", oldOptimismPortalInfo.L2Oracle, newOptimismPortalInfo.L2Oracle)
+	}
+
+	// compare SystemConfig
+	if newSystemConfigInfo.Overhead != oldSystemConfigInfo.Overhead {
```

Comment:
SystemConfig should also compare whether the value of Owner has changed.

Reply:
Added.